### ISSUE TITLE
Questions about experiment details

### ISSUE THREAD

**Issue by Pryest** [microsoft/LLM2CLIP#7]
Hello, it is surprising that the original llama retrieves almost unrelated caption, may I ask how you conduct the experiment? I failed to find more details.😂

**Comment by Yif-Yang** [comment]
> Have the same question. How do you conduct retrieval experiments on the original Llava? Are there any additional instructions or take the caption as input directly? [@Yif-Yang](https://github.com/Yif-Yang)

It seems like there might be a bit of misunderstanding. We conducted retrieval experiments on LLaMA3 not Llava, using the MS COCO 5K dataset (where each image has five captions; we used the first two captions to retrieve each other as positive samples and reported the Top-1 retrieval accuracy). We use the COCO retrieve score to measure the LLM's ability to distinguish different captions. 

For Llava1.5, we didn’t conduct retrieval tasks but rather focused on image understanding tasks, which were validated on various QA benchmarks. Could you explain specifically what you mean by "retrieval experiments on the original Llava"? 

Glad to help if you feel any confusion.

**Comment by Yif-Yang** [comment]
> > > Have the same question. How do you conduct retrieval experiments on the original Llava? Are there any additional instructions or take the caption as input directly? [@Yif-Yang](https://github.com/Yif-Yang)
> > 
> > 
> > It seems like there might be a bit of misunderstanding. We conducted retrieval experiments on LLaMA3 not Llava, using the MS COCO 5K dataset (where each image has five captions; we used the first two captions to retrieve each other as positive samples and reported the Top-1 retrieval accuracy). We use the COCO retrieve score to measure the LLM's ability to distinguish different captions.
> > For Llava1.5, we didn’t conduct retrieval tasks but rather focused on image understanding tasks, which were validated on various QA benchmarks. Could you explain specifically what you mean by "retrieval experiments on the original Llava"?
> > Glad to help if you feel any confusion.
> 
> Apologies for my misspeaking. I was referring to the original LLaMA3's caption-to-caption experiment. Could you clarify if the positive caption samples were directly used as input to LLaMA3, or if there was some prefix or instruction added to guide LLaMA3 in completing the retrieval task? Thank you for your response.

We directly use the first two captions of an image as text inputs without any prefixes. For example, in the COCO dataset with 5K images, we take the first caption of each image as text_set_1 and the second caption as text_set_2. Then, we use the 5K captions in text_set_1 as queries to retrieve captions from text_set_2. For each query, retrieval is considered correct only if the top-1 retrieved caption corresponds to the same image's text_set_2 caption. Finally, we calculate the average accuracy across all 5K queries.

Validation results show that the native LLaMA3 achieves only 5% accuracy, while CLIP's text encoder achieves 25%. After fine-tuning, LLaMA3 reaches 29%, demonstrating the importance of our fine-tuning approach. This highlights the necessity of improving feature distinguishability when using LLMs to fine-tune CLIP.

Feel free to ask more details if you still feel confused, glad to help~

**Comment by Yif-Yang** [comment]
> > > > > Have the same question. How do you conduct retrieval experiments on the original Llava? Are there any additional instructions or take the caption as input directly? [@Yif-Yang](https://github.com/Yif-Yang)
> > > > 
> > > > 
> > > > It seems like there might be a bit of misunderstanding. We conducted retrieval experiments on LLaMA3 not Llava, using the MS COCO 5K dataset (where each image has five captions; we used the first two captions to retrieve each other as positive samples and reported the Top-1 retrieval accuracy). We use the COCO retrieve score to measure the LLM's ability to distinguish different captions.
> > > > For Llava1.5, we didn’t conduct retrieval tasks but rather focused on image understanding tasks, which were validated on various QA benchmarks. Could you explain specifically what you mean by "retrieval experiments on the original Llava"?
> > > > Glad to help if you feel any confusion.
> > > 
> > > 
> > > Apologies for my misspeaking. I was referring to the original LLaMA3's caption-to-caption experiment. Could you clarify if the positive caption samples were directly used as input to LLaMA3, or if there was some prefix or instruction added to guide LLaMA3 in completing the retrieval task? Thank you for your response.
> > 
> > 
> > We directly use the first two captions of an image as text inputs without any prefixes. For example, in the COCO dataset with 5K images, we take the first caption of each image as text_set_1 and the second caption as text_set_2. Then, we use the 5K captions in text_set_1 as queries to retrieve captions from text_set_2. For each query, retrieval is considered correct only if the top-1 retrieved caption corresponds to the same image's text_set_2 caption. Finally, we calculate the average accuracy across all 5K queries.
> > Validation results show that the native LLaMA3 achieves only 5% accuracy, while CLIP's text encoder achieves 25%. After fine-tuning, LLaMA3 reaches 29%, demonstrating the importance of our fine-tuning approach. This highlights the necessity of improving feature distinguishability when using LLMs to fine-tune CLIP.
> > [@Eve1f](https://github.com/Eve1f) [@Pryest](https://github.com/Pryest) Feel free to ask more details if you still feel confused, glad to help~
> 
> Thank you for your response! You've done an outstanding job, but this experiment might lack some rationality. Perhaps providing some prompts to LlaMA3 would lead to a better experimental setup and a more convincing conlusion. 😊

Thank you for your suggestion. Here's the reasoning: **CLIP's pretraining method inherently does not use prefixes, as it relies on contrastive learning between images and captions.** Therefore, we believe that using **prefix-free** caption-to-caption accuracy **best reflects** the issue we aim to highlight—the insufficient feature distinguishability of the output layer, which fails to assist in CLIP training. This aspect might be open for further discussion, but I feel the current experiment is quite reasonable.

**Comment by Pryest** [comment]
Thx 4 response. Still think putting pure caption into LLM for text embedding not a strong baseline. Nevermind.

**Comment by Yif-Yang** [comment]
> Thx 4 response. Still think putting pure caption into LLM for text embedding not a strong baseline. Nevermind. 

Indeed, it’s true that this may not an ideal baseline for evaluating the capabilities of LLMs. However, our goal is not to evaluate the abilities of LLMs themselves. Instead, we are investigating the discriminative capabilities of different textual descriptions under the premise of how LLMs can assist CLIP. CLIP was trained without prefixed text, so we designed our experiments accordingly. Otherwise, we would have to add prefixes during CLIP’s training, which, while explored in some works, is quite unconventional and not the focus of our study.

Additionally, if we were to include prefixes for this experiment, I believe the results would remain similar. We have unpublished progress that has already validated this. That said, I believe the experimental design in our LLM2CLIP paper is appropriate for our objectives.

Thank you for your attention, and we welcome further discussion. Our goal is to contribute meaningful knowledge to the community, and your suggestions help us improve.
